pkg/controller/istio/istio_controller_test.go

```diff
@@ -303,4 +309,477 @@ var _ = Describe("Istio controller tests", func() {
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+				Spec: operatorv1.InstallationSpec{
+					Variant: "",
+				},
+			}
+			Expect(cli.Create(ctx, installationNoVariant)).NotTo(HaveOccurred())
+			Expect(cli.Create(ctx, istioCR)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			result, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(result.Requeue).To(BeTrue())
+			mockStatus.AssertCalled(GinkgoT(), "SetDegraded", operatorv1.ResourceNotReady, "Waiting for Installation Variant to be set", mock.Anything, mock.Anything)
+		})
+
+		It("should handle TigeraStatus update in reconciliation", func() {
+			createResources()
+
+			// Create TigeraStatus
+			ts := &operatorv1.TigeraStatus{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: IstioName,
+				},
+				Status: operatorv1.TigeraStatusStatus{
+					Conditions: []operatorv1.TigeraStatusCondition{
+						{
+							Type:   operatorv1.ComponentAvailable,
+							Status: operatorv1.ConditionTrue,
+						},
+					},
+				},
+			}
+			Expect(cli.Create(ctx, ts)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: IstioName}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify Istio status conditions were updated
+			updatedIstio := &operatorv1.Istio{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedIstio)).NotTo(HaveOccurred())
+			Expect(updatedIstio.Status.Conditions).NotTo(BeEmpty())
+		})
+	})
+
+	Context("Pull secrets tests", func() {
+		It("should handle pull secrets correctly", func() {
+			// Create pull secret
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "tigera-pull-secret",
+					Namespace: common.OperatorNamespace(),
+				},
+				Data: map[string][]byte{
+					".dockerconfigjson": []byte(`{"auths":{"registry.example.com":{"auth":"dGVzdDp0ZXN0"}}}`),
+				},
+				Type: corev1.SecretTypeDockerConfigJson,
+			}
+			Expect(cli.Create(ctx, secret)).NotTo(HaveOccurred())
+
+			installation.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
+				{Name: "tigera-pull-secret"},
+			}
+			Expect(cli.Create(ctx, installation)).NotTo(HaveOccurred())
+			Expect(cli.Create(ctx, istioCR)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+	})
+
+	Context("Resource rendering tests", func() {
```

Comment:
This test doesn't actually verify that resources are rendered, so I think this context is a bit of a misnomer. We should do most of the resource rendering tests in the `render` package anyway - but it would be good to use the mock client to actually check that the expected resources were created here.